recipes/_cloudwatch_agent_install.rb

```diff
@@ -0,0 +1,91 @@
+# frozen_string_literal: true
+#
+# Cookbook Name:: aws-parallelcluster
+# Recipe:: _cloudwatch_agent_install
+#
+# Copyright 2013-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with the
+# License. A copy of the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+# OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Download the cloudwatch agent's public key
+remote_file "#{node['cfncluster']['sources_dir']}/amazon-cloudwatch-agent.gpg" do
+  source node['cfncluster']['cloudwatch']['public_key_url']
+end
+
+# Download the right package based on the platform
+case node['platform']
+when 'ubuntu'
+  remote_file "#{node['cfncluster']['sources_dir']}/amazon-cloudwatch-agent.deb" do
```

Comment:
You could put the definition of a variable containing the relative path in the switch/case and have a single `remote_file` block after it,